### PR TITLE
Enable all iGEMM solvers by default

### PR DESCRIPTION
**Pull request by asroy**
This PR ~remove a~ disables workaround, so MIOpen would enable all iGEMM solvers by default.

Also resolves issue #483

### CODE REVIEW

**Comment by atamazov** [comment]
Let's discuss this on meeting today.

File: src/mlo_dir_conv.cpp [hunk 1]
```diff
@@ -255,14 +239,7 @@ std::vector<miopen::solver::ConvSolution>
 FindImplicitGemmWrWAllSolutions(const miopen::ConvolutionContext& ctx,
                                 const miopen::AnyInvokeParams& invoke_ctx)
 {
-#if WORKAROUND_SWDEV_227826
-    if(miopen::IsEnabled(MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS{}))
```
Comment (atamazov) [inline src/mlo_dir_conv.cpp:259]: If we want to reverse the default behavior, then let's simply change `IsEnabled()` to `!IsDisabled()` wherever `MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS` is used. No other changes required for now, I think.

**Review by atamazov — CHANGES_REQUESTED** [review]

File: src/mlo_dir_conv.cpp (hunk 1 above)
Comment (asroy) [reply src/mlo_dir_conv.cpp:259]: fixed

Comment (atamazov) [reply src/mlo_dir_conv.cpp:259]: Let's minimize changes.

**Review by atamazov — APPROVED** [review]
LGTM

File: src/mlo_dir_conv.cpp (hunk 1 above)
Comment (atamazov) [reply src/mlo_dir_conv.cpp:259]: Resolved by https://github.com/ROCmSoftwarePlatform/MIOpen/pull/488/commits/bbe1ffb1ac9cd4ad103405d4c0cd16d2e02f2596

**Review by JehandadKhan — APPROVED** [review]

**Comment by asroy** [comment]
@daniellowell test_conv_for_implicit_gemm failed. But there is no log on which case failed. Restarted the CI job

**Comment by atamazov** [comment]
@asroy @daniellowell Do we still need this (i.e. what is the value of it)?

**Comment by streamhsa** [comment]
blocked by #540 which include a bug fix, otherwise CI would not pass

**Comment by asroy** [comment]
@atamazov All iGEMM solvers (HIP/asm) will be used if MIOPEN_FIND_MODE=1.

This PR is blocked by #540 which has a bug fix for asm-iGEMM kernel, otherwise this PR will not pass CI

**Comment by atamazov** [comment]
@asroy 
> All iGEMM solvers (HIP/asm) will be used if MIOPEN_FIND_MODE=1.

Of course. The question is how do we estimate the overall value of this change. One may think this is not good (the value is negative), because `Find()` will take longer. Another one may consider this is of small value.

**Comment by daniellowell** [comment]
@asroy This PR needs more work. It consistently fails on FP32 gfx908 tests.
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/enable_all_igemm_solver/18/pipeline

Please open additional tickets for bug fixes.

**Comment by atamazov** [comment]
Converted to draft to avoid spoiling docker image cache on CI. You can continue work as usual. But please change this back to "normal" PR only after ending of CI moratorium (see mail). Then first merge from develop. Push the `[Ready for review]` button only after that. Thanks.

**Comment by asroy** [comment]
No longer pursuit

**Comment by atamazov** [comment]
> No longer pursuit

Why? Since #615 (2 days ago) we have parallel builds of all solutions, so the library can tolerate this.

**Pull request closed without merging**